src/oscar/apps/order/abstract_models.py

```diff
@@ -1169,3 +1177,27 @@ def description(self):
         if self.voucher_code:
             return self.voucher_code
         return self.offer_name or ""
+
+
+class AbstractSurcharge(models.Model):
+    order = models.ForeignKey(
+        'order.Order',
+        on_delete=models.CASCADE,
+        related_name="surcharges",
+        verbose_name=_("Surcharges"))
+
+    name = models.CharField(
+        _("Surcharge"), max_length=128
+    )
+
+    incl_tax = models.DecimalField(
+        _("Surcharge (inc. tax)"), decimal_places=2, max_digits=12,
+        default=0)
+
+    excl_tax = models.DecimalField(
+        _("Surcharge (excl. tax)"), decimal_places=2, max_digits=12,
+        default=0)
+
+    class Meta:
+        abstract = True
```

Comment:
Please define a default `ordering` here - otherwise surcharges will be displayed in an unpredictable order. Suggest we order by name.